src/main/java/com/monumental/controllers/SearchController.java

```diff
@@ -20,11 +23,37 @@
      * This function lets you search monuments using the q query param
      * Ex: GET http://localhost:8080/api/search?q=Memorial
      * TODO: Possibly search related tables such as Tags
-     * @param q The search query string
-     * @return  Matching Monuments based on their title
+     * @param searchQuery The search query string
+     * @return            Matching Monuments based on their title
      */
     @GetMapping("/api/search")
-    public List<Monument> searchMonuments(@RequestParam String q) {
-        return this.monumentService.search(q);
+    public List<Monument> searchMonuments(@RequestParam(required = false, value = "q") String searchQuery,
+                                          @RequestParam(required = false, defaultValue = "1", value = "page") String pageString,
+                                          @RequestParam(required = false, defaultValue = "25", value = "limit") String limitString) {
+
+        Integer page = null;
+        if (pageString != null) page = Integer.parseInt(pageString);
+        Integer limit = null;
+        if (limitString != null) limit = Integer.parseInt(limitString);
+
+        try {
+            CriteriaBuilder builder = monumentService.getCriteriaBuilder();
+            CriteriaQuery<Monument> query = monumentService.createCriteriaQuery(builder, false);
+            Root<Monument> root = monumentService.createRoot(query);
+
+            if (searchQuery != null) {
+                query.where(
+                    builder.or(
+                        builder.equal(builder.function("fts", Boolean.class, root.get("title"), builder.literal(searchQuery)), true),
+                        builder.equal(builder.function("fts", Boolean.class, root.get("artist"), builder.literal(searchQuery)), true)
+                    )
+                );
+            }
+
+            return this.monumentService.getWithCriteriaQuery(query);
```

Comment:
This is the usage of the new CriteriaBuilder stuff, which allows us to:
1. Call custom functions (in this case, "fts" which is doing the Full Text Search)
2. Build complex logic with `builder.or`, `builder.and`, etc.